src/main/java/net/mcreator/element/types/Overlay.java

```diff
@@ -58,6 +58,10 @@ public Overlay(ModElement element) {
 		this.overlayTarget = "Ingame";
 	}
 
+	public boolean hasTextures() {
+		return this.baseTexture != null && !this.baseTexture.equals("") || !getComponentsOfType("Image").isEmpty();

```

Comment:
For code clarity, please wrap `this.baseTexture != null && !this.baseTexture.equals("")` with `()`